lib/compress/zstd_opt.c

```diff
@@ -263,47 +258,80 @@ static U32 ZSTD_insertBtAndGetAllMatches (
     const U32 windowLow = zc->lowLimit;
     U32* smallerPtr = bt + 2*(current&btMask);
     U32* largerPtr  = bt + 2*(current&btMask) + 1;
-    U32 matchEndIdx = current+8;
+    U32 matchEndIdx = current+8;   /* farthest referenced position of any match => detects repetitive patterns */
     U32 dummy32;   /* to be nullified at the end */
     U32 mnum = 0;
 
-    const U32 minMatch = (mls == 3) ? 3 : 4;
     size_t bestLength = minMatchLen-1;
 
-    if (minMatch == 3) { /* HC3 match finder */
+    /* check repCode */
+    {   U32 const lastR = ZSTD_REP_NUM + ll0;
+        U32 repCode;
+        for (repCode = ll0; repCode < lastR; repCode++) {
+            U32 const repOffset = (repCode==ZSTD_REP_NUM) ? (rep[0] - 1) : rep[repCode];
+            U32 const repIndex = current - repOffset;
+            U32 repLen = 0;
+            assert(current >= dictLimit);
+            if (!extDict /*static*/ || (repIndex>=dictLimit)) {
```

Comment:
Can you add an assert here like you added earlier `assert(repIndex>=dictLimit)`? I think it will be generally useful, and will add clarity to unfamiliar readers.

Reply:
Actually, the next test `(repOffset-1 < current-dictLimit)` is equivalent to `current > repIndex >= dictLimit`. It's doing 2 tests in one comparison thanks to the overflow effect.

Which means, I don't need the preliminary `(repIndex>=dictLimit)`, since it's actually included in next test. I'll update the code (and comments) accordingly.